MmSupervisorPkg/Library/DxeMmSupervisorVersionPublicationLib/DxeMmSupervisorVersionPublicationLib.c

```diff
@@ -0,0 +1,383 @@
+/** @file
+  Publishes the MM Supervisor version information through optional interfaces.
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <PiDxe.h>
+#include <Library/BaseLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/DebugLib.h>
+#include <Library/PrintLib.h>
+#include <Library/UefiBootServicesTableLib.h>
+#include <Library/UefiLib.h>
+#include <Library/UefiRuntimeServicesTableLib.h>
+#include <Library/VariablePolicyHelperLib.h>
+
+#include <Guid/MmSupervisorRequestData.h>
+#include <Guid/MmSupervisorVersion.h>
+#include <Protocol/MmCommunication.h>
+#include <Protocol/MmSupervisorCommunication.h>
+#include <Protocol/VariablePolicy.h>
+#include <Protocol/VariableWrite.h>
+
+MM_SUPERVISOR_COMMUNICATION_PROTOCOL  *mSupvCommunicationProtocol;
+VOID                                  *mMmSupvCommonCommBufferAddress;
+UINTN                                 mMmSupvCommonCommBufferSize;
+
+/**
+  Sends the communication request to the supervisor.
+
+  @retval     EFI_SUCCESS         Communication was successful.
+  @retval     EFI_NOT_FOUND       The comm buffer or supervisor communication protocol was not found.
+  @retval     Other               The status of the supervisor communication request.
+
+**/
+EFI_STATUS
+MmSupvRequestDxeToMmCommunicate (
+  VOID
+  )
+{
+  EFI_STATUS                 Status;
+  EFI_MM_COMMUNICATE_HEADER  *CommHeader;
+  UINTN                      CommBufferSize;
+
+  if (mMmSupvCommonCommBufferAddress == NULL) {
+    DEBUG ((DEBUG_ERROR, "[%a] - Communication buffer not found!\n", __func__));
+    return EFI_NOT_FOUND;
+  }
+
+  CommHeader     = (EFI_MM_COMMUNICATE_HEADER *)mMmSupvCommonCommBufferAddress;
+  CommBufferSize = mMmSupvCommonCommBufferSize;
+
+  if (mSupvCommunicationProtocol == NULL) {
+    Status = gBS->LocateProtocol (&gMmSupervisorCommunicationProtocolGuid, NULL, (VOID **)&mSupvCommunicationProtocol);
+    if (EFI_ERROR (Status)) {
+      return EFI_NOT_FOUND;
+    }
+  }
+
+  return mSupvCommunicationProtocol->Communicate (mSupvCommunicationProtocol, CommHeader, &CommBufferSize);
+}
+
+/**
+  Locates a common communication buffer to use for interacting with the MM Supervisor.
+
+  @retval     EFI_SUCCESS         A viable comm buffer was found.
+  @retval     EFI_NOT_FOUND       The MM Supervisor Communication protocol was not found;
+
+**/
+EFI_STATUS
+LocateMmCommonCommBuffer (
+  VOID
+  )
+{
+  EFI_STATUS  Status = EFI_SUCCESS;
+
+  if (mMmSupvCommonCommBufferAddress == NULL) {
+    if (mSupvCommunicationProtocol == NULL) {
+      Status = gBS->LocateProtocol (&gMmSupervisorCommunicationProtocolGuid, NULL, (VOID **)&mSupvCommunicationProtocol);
+      if (EFI_ERROR (Status)) {
+        return EFI_NOT_FOUND;
+      }
+    }
+
+    // Use virtual start will be identical to physical start till translate event

```

Comment:
This comments wording is confusing to me.  Do you mean this?
```suggestion
    // Use virtual start which will be identical to physical start till translate event

```

Reply:
It doesn't need to be there. Removed the comment.